src/components/Reports/ReportsBox/ReportsBox.js

```diff
@@ -33,11 +33,29 @@ class ReportsBox extends React.Component {
   }
 
   componentDidMount() {
-    this.getVisitsByPeriod(this.props.domainName, this.props.periodSelectedDays);
+    this.visitsByPeriod(this.props.domainName, this.props.periodSelectedDays);
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    if (
+      props.domainName !== state.prevDomainName ||
+      props.periodSelectedDays !== state.prevPeriodSelectedDays
+    ) {
+      return {
+        visits: null,
+        prevDomainName: props.domainName,
+        prevPeriodSelectedDays: props.periodSelectedDays,
+      };
+    }
+
+    // No state update necessary
+    return null;
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.getVisitsByPeriod(nextProps.domainName, nextProps.periodSelectedDays);
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.visits === null && !this.state.isLoading) {
```

Comment:
The same here, only one flag should be enough.

Reply:
the first time visits are null if not add the other validation always enters and infinity bucle.

Reply:
mmm, that is weird, it should not happens.

There should be another hidden issue. Let me know if you want to review it together.

Reply:
Ahh, creo que me entendiste mal, yo dejaría lo sigiuente:
```suggestion
    if (this.state.visits === null) {
```

Reply:
Listo!